routing/localchans/manager.go

```diff
@@ -0,0 +1,100 @@
+package localchans
+
+import (
+	"sync"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/discovery"
+	"github.com/lightningnetwork/lnd/htlcswitch"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+)
+
+// Manager manages the node's local channels. The only operation that is
+// currently implemented is updating forwarding policies.
+type Manager struct {
+	// HtlcSwitch is used by the manager to update active links with a new
+	// policy.
+	HtlcSwitch *htlcswitch.Switch
+
+	// AuthGossiper is called to persist a new policy to disk and broadcast
+	// it to the network.
+	AuthGossiper *discovery.AuthenticatedGossiper
+
+	// Router is required to iterate over all our local channels.
+	Router *routing.ChannelRouter
```

Comment:
Not a blocker, but we could use either a slimmer interface here, or a function closure to avoid depending on the entire router struct.

Reply:
Converted to function closures and added a unit test.